lib/model/channel.dart

```diff
@@ -572,6 +641,65 @@ class ChannelStoreImpl extends HasUserStore with ChannelStore {
       forStream[event.topicName] = visibilityPolicy;
     }
   }
+
+  /// Handle a [MessageEvent], returning whether listeners should be notified.
+  bool handleMessageEvent(MessageEvent event) {
+    if (event.message is! StreamMessage) return false;
+    final StreamMessage(streamId: channelId, :topic) = event.message as StreamMessage;
+
+    final latestMessageIdsByTopic = _latestMessageIdsByChannelTopic[channelId];
+    if (latestMessageIdsByTopic == null) {
+      // We're not tracking this channel's topics yet.
+      // We start doing that when [fetchTopics] is called,
+      // and we fill in all the topics at that time.
```

Comment:
nit (and oops, I know I suggested the wording here 🙂): we don't start tracking topics when `fetchTopics` is called, but rather on success of the fetch that it performs. How about:

```dart
      // We're not tracking this channel's topics yet.
      // We'll start doing that when we get the fetched data;
      // see [fetchTopics].
```